dnsvalidator/lib/core/input.py

```diff
@@ -1,37 +1,64 @@
 #!/usr/bin/env python3
 
-import os
+import sys
 import requests
 from urllib.parse import urlparse
 from argparse import ArgumentParser
-import argparse
-
+from pathlib import Path
 
 class InputHelper(object):
     @staticmethod
     def process_targets(parser, arg):
-        targets = set()
-
-        # if target is a URL and not a file path
-        if urlparse(arg):
-            items = requests.get(arg)
-            for item in items.text.split():
-                targets.add(item)
-            return targets
-
-        if not os.path.exists(arg):
-            parser.error(
-                "The file %s does not exist or is not a valid URL!" % arg)
+        if InputHelper.validate_url(arg):
+            targets = InputHelper.process_url(parser, arg)
         else:
-            items = open(arg, 'r')
-            for item in items.strip():
-                targets.add(item.strip())
+            filename = InputHelper.validate_filename(parser, arg)
+            if filename:
+                targets = InputHelper.process_file(filename)
 
         if len(targets) == 0:
             raise Exception("No target provided, or empty target list")
 
         return targets
 
+    @staticmethod
+    def validate_url(string):
+        try:
+            result = urlparse(string)
+            # if there isn't a scheme, its probably not a URL we can reliably use
+            return result.scheme
```

Comment:
This is probably a little dodgy :D

Reply:
Still an improvement on the existing base - going to test and likely merge shortly.